### PR TITLE
docs: update jsdoc formatted comments for axe.common methods

### PR DESCRIPTION
**Pull request by akornmeier**
Also adds specific jsdoc format to: 
- axe.core.utils.contains
- axe.core.utils.findBy
- axe.core.utils.isHidden
- axe.core.utils.qsa

### CODE REVIEW

File: lib/commons/table/traverse.js [hunk 1]
```diff
@@ -24,11 +24,14 @@
 
 	/**
 	 * Traverses a table in a given direction, passing it to the callback
```
Comment (dylanb) [inline lib/commons/table/traverse.js]: `it` is a bit ambiguous

File: lib/commons/text/accessible-text-virtual.js [hunk 2]
```diff
@@ -146,6 +149,7 @@ text.accessibleText = function accessibleText(element, inLabelledByContext) {
 	return axe.commons.text.accessibleTextVirtual(virtualNode, inLabelledByContext);
 };
 
+/** @private */
```
Comment (dylanb) [inline lib/commons/text/accessible-text-virtual.js]: the virtual functions are not private - we should document these

File: lib/commons/text/label-virtual.js [hunk 3]
```diff
@@ -1,10 +1,6 @@
-/*global text, dom, axe, aria */
-/**
- * Gets the visible text of a label for a given input
- * @see http://www.w3.org/WAI/PF/aria/roles#namecalculation
- * @param  {Object} node The virtual node mapping to the input to test
- * @return {Mixed}      String of visible text, or `null` if no label is found
- */
+/* global text, dom, axe, aria */
+
+/** @private */
```
Comment (dylanb) [inline lib/commons/text/label-virtual.js]: dito

File: lib/commons/text/visible-virtual.js [hunk 4]
```diff
@@ -1,9 +1,10 @@
-/*global text, dom, axe */
+/* global text, dom, axe */
 
 /**
  * NOTE: when calculating the text or accessible text of a node that includes shadow
  * roots attached to it or its children, the flattened tree must be considered
  * rather than the "light DOM"
+ * @private
```
Comment (dylanb) [inline lib/commons/text/visible-virtual.js]: dito - won't comment on any more of the same

File: lib/commons/text/visible-virtual.js [hunk 5]
```diff
@@ -20,6 +21,13 @@ text.visibleVirtual = function (element, screenReader, noRecursing) {
 	return text.sanitize(result);
 };
 
+/**
+ * TODO: Get description and info about screenReader param...
```
Comment (dylanb) [inline lib/commons/text/visible-virtual.js]: remove todo

Comment (dylanb) [reply lib/commons/text/visible-virtual.js]: also, describe the function

File: lib/core/utils/qsa.js [hunk 6]
```diff
@@ -215,10 +210,19 @@ matchExpressions = function  (domTree, expressions, recurse) {
 	}, []);
 };
 
+/**
+ * querySelectorAll implementation that works on the virtual DOM and supports shadowDOM
```
Comment (dylanb) [inline lib/core/utils/qsa.js]: change this to `querySelectorAll implementation that operates on the flattened tree (supports shadow DOM)`

File: lib/core/utils/qsa.js [hunk 7]
```diff
@@ -215,10 +210,19 @@ matchExpressions = function  (domTree, expressions, recurse) {
...
+ * @method querySelectorAll
+ * @memberof axe.utils
+ * @instance
+ * @param  {NodeList} domTree Node collection to search
```
Comment (dylanb) [inline lib/core/utils/qsa.js]: describe domTree as `flattened tree collection to search`

File: lib/commons/color/contrast.js [hunk 8]
```diff
@@ -22,6 +23,9 @@ color.Color = function (red, green, blue, alpha) {
 
 	/**
 	 * Provide the hex string value for the color
+	 * @method toHexString
+	 * @memberof axe.commons.color.Color
```
[the remark this answers is not in the document]
Comment (akornmeier) [reply lib/commons/color/contrast.js]: It is documented as an instance on line 28

File: lib/commons/color/get-background-color.js [hunk 9]
```diff
@@ -214,6 +219,13 @@ color.getBackgroundStack = function(elm) {
 	return elmIndex !== -1 ? elmStack : null;
 };
 
+/**
+ * Returns background color for element
```
[the remark this answers is not in the document]
Comment (akornmeier) [reply lib/commons/color/get-background-color.js]: Content added.

File: lib/commons/color/element-is-distinct.js [hunk 10]
```diff
@@ -8,6 +14,15 @@ function _getFonts(style) {
 	});
 }
 
+/**
+ * Determines if compared nodes are distinct
```
[the remark this answers is not in the document]
Comment (akornmeier) [reply lib/commons/color/element-is-distinct.js]: updated.

File: lib/commons/color/get-background-color.js [hunk 11]
```diff
@@ -182,6 +182,11 @@ function sortPageBackground(elmStack) {
 /**
  * Get all elements rendered underneath the current element,
  * in the order in which it is rendered
```
[the remark this answers is not in the document]
Comment (akornmeier) [reply lib/commons/color/get-background-color.js]: updated

File: lib/commons/dom/get-composed-parent.js [hunk 12]
```diff
@@ -1,8 +1,11 @@
-/*global dom */
+/* global dom */
 /**
  * Get an element's parent in the composed tree
```
[the remark this answers is not in the document]
Comment (akornmeier) [reply lib/commons/dom/get-composed-parent.js]: ☑️ Done

File: lib/commons/dom/get-element-by-reference.js [hunk 13]
```diff
@@ -1,5 +1,11 @@
 /*global dom */
 
+/**
+ * Finds the first Element with the given attribute
```
[the remark this answers is not in the document]
Comment (akornmeier) [reply lib/commons/dom/get-element-by-reference.js]: ☑️ Done

**Comment by WilcoFiers** [comment]
@akornmeier What's the status of this?

File: lib/commons/dom/get-composed-parent.js [hunk 14]
```diff
@@ -1,8 +1,11 @@
-/*global dom */
+/* global dom */
```
Comment (isner) [inline lib/commons/dom/get-composed-parent.js]: My OCD thanks you a thousand times, @akornmeier

File: lib/commons/utils/token-list.js [hunk 15]
```diff
@@ -1,7 +1,14 @@
...
+/* global axe */
+/**
+ * Converts space delimited token list to an Array
+ * @method tokenList
+ * @memberof axe.commons.utils
+ * @instance
+ * @param  {String} str
+ * @return {Array}
+ */
 axe.utils.tokenList = function (str) {
```
Comment (isner) [inline lib/commons/utils/token-list.js]: It's out of scope for this PR, but I don't find this utility's name to be particularly intuitive. It speaks to nothing of what it does. Does it create a token list? Process one? Get one? `tokenListToArray` would be nice.

**Pull request merged**